### ISSUE TITLE
Weird zebra ghost in the generated image background for horse2zebra

### ISSUE THREAD

**Issue by ethanyanjiali** [junyanz/pytorch-CycleGAN-and-pix2pix#656]
After 200 epochs training, some of the generated data looks like this:

Does anyone know what could lead to this and what would be a good way to improve? 

![a2b_8_original](https://user-images.githubusercontent.com/8679679/58442544-f4b2ec00-80a0-11e9-97d4-9104687e5142.JPEG)
![a2b_8_generated](https://user-images.githubusercontent.com/8679679/58442545-f67caf80-80a0-11e9-99e8-f0d7c767d2a4.JPEG)
![a2b_16_original](https://user-images.githubusercontent.com/8679679/58442559-0b594300-80a1-11e9-90b9-07fb970b4754.JPEG)
![a2b_16_generated](https://user-images.githubusercontent.com/8679679/58442561-0d230680-80a1-11e9-882d-8d666cd1d9ae.JPEG)
![a2b_33_original](https://user-images.githubusercontent.com/8679679/58442572-1a3ff580-80a1-11e9-92d4-f9df7faa0dd7.JPEG)
![a2b_33_generated](https://user-images.githubusercontent.com/8679679/58442577-1ca24f80-80a1-11e9-81d5-6b7cfb9436a2.JPEG)
![3_a2b_test_original_0](https://user-images.githubusercontent.com/8679679/58442584-25932100-80a1-11e9-97ab-679468c11a8d.JPEG)
![3_a2b_test_generated_0](https://user-images.githubusercontent.com/8679679/58442596-2af06b80-80a1-11e9-845c-eb2d686236e0.JPEG)

**Comment by ethanyanjiali** [comment]
And i feel like the identity loss won't address this issue effectively because it reconstructed image does erased the ghost in the background:

original:
![3_a2b_test_original_0](https://user-images.githubusercontent.com/8679679/58442695-bb2eb080-80a1-11e9-8741-7bb32bf9bf39.JPEG)

generated:
![b2a_0_original](https://user-images.githubusercontent.com/8679679/58442705-c97ccc80-80a1-11e9-890c-9542376777aa.JPEG)

reconstructed:
![b2a_0_generated](https://user-images.githubusercontent.com/8679679/58442707-cf72ad80-80a1-11e9-9886-99a6e2e6e28d.JPEG)

**Comment by junyanz** [comment]
Does it work for the pre-trained horse2zebra model? @taesungp

**Comment by code-Assasin** [comment]
I am facing the problem with my own dataset .. any solutions?

**Comment by junyanz** [comment]
You can use a larger lambda for identity loss and cycle-consistenty loss.  Another trick is to use smaller patches for training.
